### DISCUSSION TITLE
difference between a requirement and a spec

### DISCUSSION THREAD

**Discussion by sembooooo** [useblocks/sphinx-needs#255] [Q&A] [answered]
Hi People,
This is more of a definitions related question
I was going through the documentation of sphinx-needs and got the following doubt.
I noticed two things in the sphinx-needs documentation one is a requirement(req) and the second one is specification (spec)
What are the difference between these two ?
When to use what ?

**_Why i got this doubt:_**
Normally a specification according to my understanding is a collection of requirements.
But here it is a little different. 
So What is thought process behind creating such different different things ?

Are there any definitions to those (req, spec, feature etc.,.) ? 
If there are definitions already then could you please direct me to the link where they are written?

Thank you for you answers,

**Comment by danwos** [comment, marked as answer]
Hi @sembooooo and welcome to Sphinx-Needs,
first of all,  Sphinx-Needs does not follow / requires any specific process and it can be configured to support any process with any kind of process objects (req, specs, stories, features, bugs, enhancements, tasks, ...).

However, there are some default need types defined to make the start easier (req, spec, impl, test).
A spec contains the technical details of how to reach a specific requirement.
The requirement normally describes the functional point, without any technical details.
The spec then tells the developer how this needs to get implement in the normally company specific product.
But each team/company may define these things a little different.
See example below.

But however your own definition is, it is not important for Sphinx-Needs itself, as it does only the technical management of these objects.
So you are free to use Sphinx-Needs however you want to use it.

Small hint: Most of my projects normally have some kind of a  "How-To" page in the documentation, which exactly describes how the terms and need types shall be used.




### Example (rough)
**req**: User needs to login to use our web service
**spec 1**: Create a login screen with bootstrap and co.
**spec 2**: Store this user class in our MongoDB
**spec 3**:  Use login-solution XY for authentication
